### PR TITLE
export more interfaces, fix compile problem

### PR DESCRIPTION
**Pull request by yiwenxue**
export more interfaces, fix compile problem

1. export setlimit/getlimit for hinge joint.
2. export interfaces for d6f joint.
3. fix wasm build problem `calling protected destructor`.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@cocos/physx",
-  "version": "1.0.8",
+  "version": "1.0.9-beta.d6joint.4",
```
Comment (lealzhan) [inline package.json]: ?

File: .npmignore [hunk 2]
```diff
@@ -9,7 +9,9 @@ pxshared
 *.sh
 *.bat
 compile_commands.json
+test
+
+emsdk
 
 builds/physx.checked.*
 builds/physx.profile.*
-builds/physx.debug.*
```
Comment (lealzhan) [inline .npmignore:15]: why remove builds/physx.debug.* ?

File: physx/source/compiler/cmake/emscripten/CMakeLists.txt [hunk 3]
```diff
@@ -28,7 +28,7 @@
...
...ARNINGS "-ferror-limit=0 -Wall -Wextra -Werror -Wstrict-aliasing=2  -Weverything -Wno-double-promotion -Wno-documentation-deprecated-sync -Wno-documentation-unknown-command -Wno-gnu-anonymous-struct -Wno-undef -Wno-unused-function -Wno-nested-anon-types -Wno-float-equal -Wno-padded -Wno-weak-vtables -Wno-cast-align -Wno-conversion -Wno-missing-noreturn -Wno-missing-variable-declarations -Wno-shift-sign-overflow -Wno-covered-switch-default -Wno-exit-time-destructors -Wno-global-constructors -Wno-missing-prototypes -Wno-unreachable-code -Wno-unused-macros -Wno-unused-member-function -Wno-used-but-marked-unused -Wno-weak-template-vtables -Wno-deprecated -Wno-non-virtual-dtor -Wno-invalid-noreturn -Wno-return-type-c-linkage -Wno-reserved-id-macro -Wno-c++98-compat-pedantic -Wno-unused-local-typedef -Wno-old-style-cast -Wno-newline-eof -Wno-unused-private-field -Wno-format-nonliteral -Wno-implicit-fallthrough -Wno-undefined-reinterpret-cast -Wno-disabled-macro-expansion -Wno-zero-as-null-pointer-constant -Wno-shadow -Wno-unknown-warning-option -Wno-atomic-implicit-seq-cst -Wno-extra-semi-stmt -Wno-dollar-in-identifier-extension -Wno-alloca -Wno-anon-enum-enum-conversion -Wno-dtor-name")
```
Comment (lealzhan) [inline physx/source/compiler/cmake/emscripten/CMakeLists.txt:31]: why adds -Wno-double-promotion ?

**Review by lealzhan — APPROVED** [review]

File: physx/include/characterkinematic/PxExtended.h [hunk 4]
```diff
@@ -49,7 +49,7 @@ namespace physx
 #define PX_BIG_WORLDS
 
 #ifdef PX_BIG_WORLDS
-typedef	double	PxExtended;
+typedef	float	PxExtended;
```
Comment (dumganhar) [inline physx/include/characterkinematic/PxExtended.h:52]: Why change the double to float for PxExtended?

**Pull request merged**